### PR TITLE
Remove stubbing in settings integration specs

### PR DESCRIPTION
**Pull request by imtayadeway**
There was a lot of stubbing of the SUT in the settings integration specs that were added to cover theoretical future edge cases in the settings whitelist configuration. By separating the concerns of slicing up settings and filtering them according to user role/whitelist, we are able to test them in isolation, removing the need for stubbed integration tests.

Fixes https://github.com/ManageIQ/manageiq-api/issues/270

### CODE REVIEW

File: lib/services/api/settings_slicer.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Api
+  class SettingsSlicer
+    def self.slice(settings, *path)
```
[the remark this answers is not in the document]
Comment (imtayadeway) [reply lib/services/api/settings_slicer.rb]: I'm open to the idea of this small service living elsewhere, but I don't think that SettingsFilterer is the right place for it since it's a service that has a single responsibility.

I personally see small classes that do one thing as a design goal, not a cost that needs to be mitigated :grin:

**Comment by abellotti** [comment]
Thanks @imtayadeway for the update, this is looking good.  I was expecting though that the settings support we have for servers/:id/settings, regions/:id/settings and zones/:id/settings to also go through the SettingsFilterer, i.e. the common code in app/controllers/api/base_controller.rb.  Can you update this here or you prefer another PR ?  Thanks.

**Comment by imtayadeway** [comment]
@abellotti that sounds like a great idea! I'll definitely do that, let's make it a follow up to prevent this PR getting too large. Thanks!
